Surface_mesh/include/CGAL/draw_surface_mesh.h

```diff
@@ -112,33 +112,35 @@ struct Graphics_scene_options_surface_mesh
 
   Graphics_scene_options_surface_mesh(const SM& amesh)
   {
-    bool found=false;
-    std::tie(vcolors, found)=
+    std::optional<typename SM::template Property_map<vertex_descriptor, CGAL::IO::Color>> _vcolors =
         amesh.template property_map<vertex_descriptor, CGAL::IO::Color>("v:color");
-    if(found)
+    if(_vcolors.has_value())
     {
+      vcolors = _vcolors.value();
       this->colored_vertex=[](const SM &, vertex_descriptor)->bool { return true; };
       this->vertex_color=[this](const SM &, vertex_descriptor v)->CGAL::IO::Color
       { return get(vcolors, v); };
     }
     else
     { this->colored_vertex=[](const SM &, vertex_descriptor)->bool { return false; }; }
 
-    std::tie(ecolors, found)=
-        amesh.template property_map<edge_descriptor, CGAL::IO::Color>("e:color");
-    if(found)
+    std::optional<typename SM::template Property_map<edge_descriptor, CGAL::IO::Color>> _ecolors
+      = amesh.template property_map<edge_descriptor, CGAL::IO::Color>("e:color");
+    if(_ecolors.has_value())
     {
+      ecolors = _ecolors.value();
       this->colored_edge=[](const SM &, edge_descriptor)->bool { return true; };
       this->edge_color=[this](const SM &, edge_descriptor e)->CGAL::IO::Color
       { return get(ecolors, e); };
     }
     else
     { this->colored_edge=[](const SM &, edge_descriptor)->bool { return false; }; }
 
-    std::tie(fcolors, found)=
+    auto _fcolors =
         amesh.template property_map<face_descriptor, CGAL::IO::Color>("f:color");
-    if(found)
+    if(_fcolors.has_value())
     {
+      fcolors = _fcolors.value();
```

Comment:
There should be a consistent use of `auto` versus the full type. I prefer auto, by the way, because it will always avoid type casting.

Reply:
But that is not a reason to delay the merge of the PR.